client/wsclient_test.go

```diff
@@ -177,3 +180,50 @@ func TestVerifyWSCompress(t *testing.T) {
 		})
 	}
 }
+
+func redirectServer(to string) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		http.Redirect(w, req, to, http.StatusSeeOther)
+	}))
+}
+
+func TestRedirectWS(t *testing.T) {
```

Comment:
Can you add another test case that return 3xx response but does not include "Location" to test for the error path?

Reply:
See coverage report: https://app.codecov.io/github/open-telemetry/opamp-go/commit/353be65d240a6e5129d9d2cb4fd10f1434489aca/blob/client/wsclient.go#L135

It would be nice to cover the paths that are not covered.